service-base/src/main/java/org/eclipse/hono/service/AbstractServiceBase.java

```diff
@@ -40,6 +44,9 @@
  */
 public abstract class AbstractServiceBase<T extends ServiceConfigProperties> extends ConfigurationSupportingVerticle<T> implements HealthCheckProvider {
 
+    // TODO make configurable
+    protected static final long HEARTBEAT_RATE_MILIIS = HealthCheckServer.LIVENESS_TIMEOUT_SECONDS * 500;
```

Comment:
The idea is to send updates more frequently than the timeout to prevent small delays from causing failing liveness checks. Do you object this idea or just need the reasoning to be more explicit?

Reply:
I am not sure if I am convinced of the liveness check at all ;-) In the end, there usually is not much to check. If the VM has crashed, then there will be no response to the external liveness check requests anyway. The only reasonable liveness check that we have implemened so far is one for checking if the event loop is blocked. But in that case the test itself runs for several seconds (waiting for the timeout to occur) so in that case it doesn't seem to make much sense to set the liveness timeout to say 2 secs and run the check every second, does it?

Reply:
The proposal for the liveness check in this PR is to send a "heartbeat" from each verticle instance and if it stops arriving in the expected time frame at the `HealthCheckServer` the verticle instance is assumed to be "dead".  As I understand it, this could replace the current check for a blocked event loop. 
We could take the value for the maximal time frame after a verticle instance is considered "dead" for example from the Kubernetes configuration of `livenessProbe.periodSeconds` (which defaults to 10 seconds). The heartbeat needs to be sent more frequently than this maximal time frame to prevent that a small delay, e.g. on the event bus, triggers a failing liveness check.
The idea of the heartbeat is optional. Alternatively we could regularly schedule the current check for a blocked event loop.